modules/tensor_mechanics/include/materials/RadialReturnStressUpdate.h

```diff
@@ -188,4 +245,60 @@ class RadialReturnStressUpdate : public StressUpdateBase, public SingleVariableR
+                                                             bool compute_full_tangent_operator,
+                                                             RankFourTensor & tangent_operator);
+
+template <>
+void RadialReturnStressUpdateTempl<false>::updateState(RankTwoTensor & strain_increment,
+                                                       RankTwoTensor & inelastic_strain_increment,
+                                                       const RankTwoTensor & rotation_increment,
+                                                       RankTwoTensor & stress_new,
+                                                       const RankTwoTensor & stress_old,
+                                                       const RankFourTensor & elasticity_tensor,
+                                                       const RankTwoTensor & elastic_strain_old,
+                                                       bool compute_full_tangent_operator,
+                                                       RankFourTensor & tangent_operator);
+
+template <>
+void RadialReturnStressUpdateTempl<true>::updateState(ADRankTwoTensor & strain_increment,
+                                                      ADRankTwoTensor & inelastic_strain_increment,
+                                                      const ADRankTwoTensor & rotation_increment,
+                                                      ADRankTwoTensor & stress_new,
+                                                      const RankTwoTensor & stress_old,
+                                                      const ADRankFourTensor & elasticity_tensor,
+                                                      const RankTwoTensor & elastic_strain_old);
+
+template <>
+void
+RadialReturnStressUpdateTempl<false>::updateStateSubstep(RankTwoTensor & strain_increment,
+                                                         RankTwoTensor & inelastic_strain_increment,
+                                                         const RankTwoTensor & rotation_increment,
+                                                         RankTwoTensor & stress_new,
+                                                         const RankTwoTensor & stress_old,
+                                                         const RankFourTensor & elasticity_tensor,
+                                                         const RankTwoTensor & elastic_strain_old,
+                                                         bool compute_full_tangent_operator,
+                                                         RankFourTensor & tangent_operator);
+
+template <>
+void RadialReturnStressUpdateTempl<true>::updateStateSubstep(
+    ADRankTwoTensor & strain_increment,
+    ADRankTwoTensor & inelastic_strain_increment,
+    const ADRankTwoTensor & rotation_increment,
+    ADRankTwoTensor & stress_new,
+    const RankTwoTensor & stress_old,
+    const ADRankFourTensor & elasticity_tensor,
+    const RankTwoTensor & elastic_strain_old);
```

Comment:
why are these required?

Reply:
The declarations of the specializations are needed here to prevent calling the default templated function. Removing these, the default function that errors out will be called.